server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -1111,12 +950,7 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
                                         result.v1().getGenId(), null, Priority.IMMEDIATE, listener));
                     },
                     e -> {
-                        if (abortedDuringInit) {
```

Comment:
No need to wait here anymore. If the snapshot was aborted during INIT, then that means it was already simply removed from the CS so no point in waiting here. This only made sense when we had the INIT state and would move `INIT` -> `ABORT` -> remove via https://github.com/elastic/elasticsearch/pull/55918/files#diff-a0853be4492c052f24917b5c1464003dL424 (or the apply CS method) which isn't a thing any longer without the INIT state.